### PR TITLE
OSX port + CMake project

### PR DESCRIPTION
**Pull request by JoakimSoderberg**
I ported this to OSX and created a CMake project as well for compiling.

Shouldn't be super hard to port it to windows as well... using [LoadLibrary](https://msdn.microsoft.com/en-us/library/ms684175(VS.85).aspx) instead of `dlopen` stuff.

Also fixed *some* memory leaks, there are more. As well as illegal writes... Didn't have time to fix them all.

### CODE REVIEW

File: src/nmea/parser.c [hunk 1]
```diff
@@ -42,7 +53,17 @@ _get_so_files(const char *path, char **files)
...
 
+	if (d)
+		closedir(d);
+
 	return i;
+fail:
+	for (j = 0; j < i; i++) {
+		free(files[j]);
+	}
+	if (d)
```
Comment (jacketizer) [inline src/nmea/parser.c]: if (d) {

File: src/nmea/parser.c [hunk 2]
```diff
@@ -42,7 +53,17 @@ _get_so_files(const char *path, char **files)
...
+		closedir(d);
```
Comment (jacketizer) [inline src/nmea/parser.c]: Add a '}' to the line below.

**Comment by jacketizer** [comment]
Thank you, really nice fixes :+1: I posted some very small comments, if you want, I can merge and fix them my self.

**Comment by JoakimSoderberg** [comment]
@jacketizer they're already fixed :)

**Comment by jacketizer** [comment]
Wow, that is fast :D

**Pull request merged**